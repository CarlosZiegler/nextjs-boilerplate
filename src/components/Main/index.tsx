import React from 'react'
import * as S from './styles'

function Main() {
  return (
    <S.Wrapper>
      <S.Logo
        src="/img/logo.svg"
        alt="Imagem de um atomo e React Avancado escrito ao lado"
      />
      <S.Title>React Avan√ßado</S.Title>
      <S.Description>
        Typescript, ReactJs, NextJs e Styled Componets
      </S.Description>
      <S.Illustration src="/img/hero-illustration.svg" />
    </S.Wrapper>
  )
}

export default Main
