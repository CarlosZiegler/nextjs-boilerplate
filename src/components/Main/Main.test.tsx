import { render, screen } from '@testing-library/react'
import Main from '.'

describe('loads items eventually', () => {
  it('loads items eventually', async () => {
    const { container } = render(<Main />)

    expect(
      screen.getByRole('heading', { name: /react avan√ßado/i })
    ).toBeInTheDocument()

    expect(container.firstChild).toMatchSnapshot()
  })
})
