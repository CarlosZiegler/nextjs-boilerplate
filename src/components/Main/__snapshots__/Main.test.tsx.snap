// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`loads items eventually loads items eventually 1`] = `
<main>
  <h1>
    React Avançado
  </h1>
</main>
`

exports[
  `loads items eventually loads items eventually: 
      <main>
        <h1>
          React Avançado
        </h1>
      </main>
     1`
] = `
<main>
  <h1>
    React Avançado
  </h1>
</main>
`
